src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -62,34 +63,95 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
-  /** Stores the user in Datastore (if not already in it) */
-  public void storeUser(String user) {
-    Entity userEntity = new Entity("User", user);
-    if (!userFound(user)) {
+  /**
+   * Either stores the user in datastore, or updates the messagesSent
+   * property when a message is being stored.
+   */
+  public void storeUser(String username, long messagesSent) {
+    Entity userEntity = new Entity("User", username);
+    userEntity.setProperty("messagesSent", messagesSent);
+    // Storing a user when they first log in.
+    if (messagesSent == 0) {
+      if (!userFound(username)) {
+        datastore.put(userEntity);
```

Comment:
Why would you need !userFound(username)?  Why not always save?

Reply:
This is called when a user logs in every time. We end up overwriting the messagesSent property if we always put in the datastore from there.